artichoke-backend/src/extn/core/time/offset.rs

```diff
@@ -0,0 +1,211 @@
+//! Parser for Ruby Time offset parameter to help generate `Time`.
+
+use crate::convert::{implicitly_convert_to_int, implicitly_convert_to_string};
+use crate::extn::core::symbol::Symbol;
+use crate::extn::core::time::Offset;
+use crate::extn::prelude::*;
+
+#[allow(clippy::cast_precision_loss)]
+const MAX_FLOAT_OFFSET: f64 = i32::MAX as f64;
+#[allow(clippy::cast_precision_loss)]
+const MIN_FLOAT_OFFSET: f64 = i32::MIN as f64;
+
+impl TryConvertMut<Value, Option<Offset>> for Artichoke {
+    type Error = Error;
+
+    fn try_convert_mut(&mut self, options: Value) -> Result<Option<Offset>, Self::Error> {
+        let hash: Vec<(Value, Value)> = self.try_convert_mut(options)?;
+
+        // Short circuit. An empty options hash does not error.
+        //
+        // Example:
+        //
+        // ```console
+        // [2.6.3]> Time.at(0, {})
+        // => 1970-01-01 01:00:00 +0100
+        // ```
+        if hash.is_empty() {
+            return Ok(None);
+        }
+
+        // All other keys are rejected.
+        //
+        // Example:
+        // ```console
+        // [2.6.3]> Time.at(0, i: 0)
+        // ArgumentError (unknown keyword: i)
+        // ```
+        for (mut key, _) in &hash {
+            let k = unsafe { Symbol::unbox_from_value(&mut key, self)? }.bytes(self);
```

Comment:
it looks like we need to handle non-`Symbol` args in the same way:

```
[3.1.2] > Time.at(12, 'in' => 'Z')
<internal:timev>:270:in `at': unknown keyword: "in" (ArgumentError)
        from (irb):1:in `<main>'
        from /usr/local/var/rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `load'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `<main>'
[3.1.2] > Time.at(12, Object.new => 'Z')
<internal:timev>:270:in `at': unknown keyword: #<Object:0x00000001091c7d70> (ArgumentError)
        from (irb):2:in `<main>'
        from /usr/local/var/rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `load'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `<main>'
```

Reply:
Filed https://github.com/artichoke/artichoke/issues/2092.

This is related to https://github.com/artichoke/artichoke/issues/2091.